.github/workflows/member_pr.yml

```diff
@@ -0,0 +1,59 @@
+### .github/workflows/member_pr.yml
+### This workflow operates on the target branch, which is "main," ensuring that no pull request can alter the logic until it has been merged into the "main" branch.
+name: Membership check
+on:
+    # It's important to use "pull_request_target" here to prevent potentially malicious users from creating pull requests and bypassing the constraint.
+    pull_request_target:
+        types: [opened, synchronize]
+        paths:
+          - mimir-build-image/Dockerfile 
+        branches:
+          - main
+          - 'release-**'
+
+jobs:
+    check-membership:
+        runs-on: ubuntu-latest
+        steps:
+          - name: Check out repository
+            uses: actions/checkout@v4
+          - name: Check user for team affiliation
+            id: teamAffiliation
+            run: |
+                # Encode the triggering actor's username for use in the GitHub API URL. Otherwise the user like renovate[bot] has encoding error
+                ENCODED_USERNAME=$(printf %s "${{ github.triggering_actor }}" | xxd -p -c 256 | tr -d '\n' | sed 's/\(..\)/%\1/g')
+                
+                # Define the GitHub API URL for checking team membership.
+                API_URL="https://api.github.com/orgs/grafana/teams/mimir-maintainers/members/$ENCODED_USERNAME"
+                
+                # Send a GET request to the GitHub API with proper authentication.
+                response=$(curl -s -o /dev/null -w "%{http_code}" -H "Authorization: token ${{ secrets.GH_BOT_ACCESS_TOKEN }}" $API_URL)
```

Comment:
Could you instead use `gh api orgs/grafana/teams/mimir-maintainers/members/${{ github.triggering_actor }}` here? Ought to be much simpler than going via `curl`.